From c9160774740e802c678acaa9602400ea5c506304 Mon Sep 17 00:00:00 2001
From: FanBin <fanbin12@huawei.com>
Date: Tue, 21 Mar 2023 10:23:27 +0800
Subject: [PATCH] lwip send/recv thread bind numa

---
 src/include/lwipsock.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/include/lwipsock.h b/src/include/lwipsock.h
index 0ae5896..769644f 100644
--- a/src/include/lwipsock.h
+++ b/src/include/lwipsock.h
@@ -139,6 +139,7 @@ struct lwip_sock {
   const struct rte_memzone *same_node_rx_ring_mz;
   struct same_node_ring *same_node_tx_ring;
   const struct rte_memzone *same_node_tx_ring_mz;
+  uint8_t already_bind_numa;
 #endif
 };
 
-- 
2.33.0

